client/client.go

```diff
@@ -17,40 +15,40 @@ import (
 func main() {
 	log.Println("Client started...")
 
-	tlsFilePath := os.Getenv("RATLS_FILE_PATH")
-	tlsCertPath := "tls/tlscert.der"
-	tlsKeyPath := "tls/tlskey.der"
-
-	if tlsFilePath != "" {
-		tlsCertPath = path.Join(tlsFilePath, "tlscert.der")
-		tlsKeyPath = path.Join(tlsFilePath, "tlskey.der")
-	}
-
 	// create TLS certificate and key
 	err := ratls_wrapper.LoadRATLSLibs()
 	if err != nil {
 		log.Fatal(err.Error())
 	}
 
-	err = ratls_wrapper.RATLSCreateKeyAndCrtDer(tlsCertPath, tlsKeyPath)
+	derKey, derCert, err := ratls_wrapper.RATLSCreateKeyAndCrtDer()
 	if err != nil {
 		log.Fatal(err.Error())
 	}
 
 	// Read the key pair to create certificate
-	cert, err := mutual_ratls.LoadX509KeyPairDER(tlsCertPath, tlsKeyPath)
+	cert, err := mutual_ratls.X509KeyPairDER(derKey, derCert)
 	if err != nil {
 		log.Fatal(err)
 	}
 
+	mrenclave := mutual_ratls.GetSGXEnvVar("MRENCLAVE")
+	if mrenclave == nil {
+		log.Fatal("required mrenclave for enclave measurement.")
```

Comment:
Why fail if mrenclave is nil?

Reply:
I misunderstood that verifying mrenclave is required. will change this to optional.